kotlin-sdk/src/commonMain/kotlin/dev/openfeature/kotlin/sdk/multiprovider/MultiProvider.kt

```
@@ -177,73 +171,141 @@ class MultiProvider(
+    /**
+     * Runs one of [this] child's lifecycle calls, leaving the failure to the child: it reports its
+     * own error event, so one failing child must not cancel the siblings.
+     */
+    private suspend fun ChildFeatureProvider.reportingItsOwnFailure(
+        work: suspend ChildFeatureProvider.() -> Unit
+    ) {
+        try {
+            work()
+        } catch (e: CancellationException) {
+            throw e
+        } catch (e: Throwable) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)